pkg/controllers/state/cluster.go

```diff
@@ -269,14 +273,15 @@ func (c *Cluster) UpdateNode(ctx context.Context, node *v1.Node) error {
 	}
 	c.nodes[node.Spec.ProviderID] = n
 	c.nodeNameToProviderID[node.Name] = node.Spec.ProviderID
+	ClusterStateNodesTotal.WithLabelValues().Set(float64(len(c.nodes)))
```

Comment:
I'm imagining that we should also update this on updates to NodeClaims or this  is going to be a little behind. StateNodes will get created as soon as a NodeClaim is created

Reply:
Noted, moved